improver/synthetic_data/generate_metadata.py

```diff
@@ -193,6 +194,11 @@ def generate_metadata(
     data = _create_data_array(
         ensemble_members, leading_dimension, npoints, kwargs["height_levels"]
     )
+    if "attributes" not in kwargs:
```

Comment:
I'm not sure this is the correct solution to this problem. I think that if these attributes are mandatory for any improver cube then we should fail if none are provided and make it clear that it's required that they are passed in (and pass them into the relevant acceptance tests). I think this is what Caroline was getting at in her notebook. What do you think? Do we need to discuss?

Reply:
As discussed, I've changed the behaviour to fail if any of these mandatory attributes are not set. The plugin and CLI now require a dict or JSON file to satisfy this.